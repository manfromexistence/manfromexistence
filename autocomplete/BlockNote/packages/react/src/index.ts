// TODO: review directories
export * from "./editor/BlockNoteContext";
export * from "./editor/BlockNoteDefaultUI";
export * from "./editor/BlockNoteTheme";
export * from "./editor/BlockNoteView";
export * from "./editor/defaultThemes";

export * from "./components/FormattingToolbar/FormattingToolbarController";
export * from "./components/FormattingToolbar/FormattingToolbarProps";
export * from "./components/FormattingToolbar/mantine/DefaultButtons/BasicTextStyleButton";
export * from "./components/FormattingToolbar/mantine/DefaultButtons/ColorStyleButton";
export * from "./components/FormattingToolbar/mantine/DefaultButtons/CreateLinkButton";
export * from "./components/FormattingToolbar/mantine/DefaultButtons/ImageCaptionButton";
export * from "./components/FormattingToolbar/mantine/DefaultButtons/NestBlockButtons";
export * from "./components/FormattingToolbar/mantine/DefaultButtons/ReplaceImageButton";
export * from "./components/FormattingToolbar/mantine/DefaultButtons/TextAlignButton";
export * from "./components/FormattingToolbar/mantine/DefaultSelects/BlockTypeSelect";
export * from "./components/FormattingToolbar/mantine/FormattingToolbar";

export * from "./components/LinkToolbar/LinkToolbarController";
export * from "./components/LinkToolbar/LinkToolbarProps";
export * from "./components/LinkToolbar/mantine/DefaultButtons/DeleteLinkButton";
export * from "./components/LinkToolbar/mantine/DefaultButtons/EditLinkButton";
export * from "./components/LinkToolbar/mantine/DefaultButtons/OpenLinkButton";
export * from "./components/LinkToolbar/mantine/LinkToolbar";
export * from "./components/LinkToolbar/mantine/EditLinkMenuItems";

export * from "./components/SideMenu/SideMenuController";
export * from "./components/SideMenu/SideMenuProps";
export * from "./components/SideMenu/mantine/DefaultButtons/AddBlockButton";
export * from "./components/SideMenu/mantine/DefaultButtons/DragHandleButton";
export * from "./components/SideMenu/mantine/SideMenu";
export * from "./components/SideMenu/mantine/SideMenuButton";

export * from "./components/SideMenu/DragHandleMenu/DragHandleMenuProps";
export * from "./components/SideMenu/DragHandleMenu/mantine/DefaultItems/BlockColorsItem";
export * from "./components/SideMenu/DragHandleMenu/mantine/DefaultItems/RemoveBlockItem";
export * from "./components/SideMenu/DragHandleMenu/mantine/DragHandleMenu";
export * from "./components/SideMenu/DragHandleMenu/mantine/DragHandleMenuItem";

export * from "./components/SuggestionMenu/SuggestionMenuController";
export * from "./components/SuggestionMenu/SuggestionMenuWrapper";
export * from "./components/SuggestionMenu/getDefaultReactSlashMenuItems";
export * from "./components/SuggestionMenu/hooks/useCloseSuggestionMenuNoItems";
export * from "./components/SuggestionMenu/hooks/useLoadSuggestionMenuItems";
export * from "./components/SuggestionMenu/hooks/useSuggestionMenuKeyboardNavigation";
export * from "./components/SuggestionMenu/mantine/SuggestionMenu";
export * from "./components/SuggestionMenu/mantine/SuggestionMenuItem";
export * from "./components/SuggestionMenu/types";

export * from "./components/ImagePanel/ImagePanelController";
export * from "./components/ImagePanel/ImagePanelProps";
export * from "./components/ImagePanel/mantine/DefaultTabs/EmbedTab";
export * from "./components/ImagePanel/mantine/DefaultTabs/UploadTab";
export * from "./components/ImagePanel/mantine/ImagePanel";
export * from "./components/ImagePanel/mantine/ImagePanelButton";
export * from "./components/ImagePanel/mantine/ImagePanelFileInput";
export * from "./components/ImagePanel/mantine/ImagePanelTab";
export * from "./components/ImagePanel/mantine/ImagePanelTextInput";

export * from "./components/TableHandles/TableHandlesController";
export * from "./components/TableHandles/TableHandleProps";
export * from "./components/TableHandles/hooks/useTableHandlesPositioning";
export * from "./components/TableHandles/mantine/TableHandle";

export * from "./components/TableHandles/TableHandleMenu/TableHandleMenuProps";
export * from "./components/TableHandles/TableHandleMenu/mantine/DefaultButtons/AddButton";
export * from "./components/TableHandles/TableHandleMenu/mantine/DefaultButtons/DeleteButton";
export * from "./components/TableHandles/TableHandleMenu/mantine/TableHandleMenu";
export * from "./components/TableHandles/TableHandleMenu/mantine/TableHandleMenuItem";

export * from "./components/mantine-shared/Toolbar/ToolbarButton";
export * from "./components/mantine-shared/Toolbar/ToolbarSelect";

export * from "./hooks/useActiveStyles";
export * from "./hooks/useBlockNoteEditor";
export * from "./hooks/useCreateBlockNote";
export * from "./hooks/useEditorChange";
export * from "./hooks/useEditorContentOrSelectionChange";
export * from "./hooks/useEditorForceUpdate";
export * from "./hooks/useEditorSelectionChange";
export * from "./hooks/useSelectedBlocks";

export * from "./schema/ReactBlockSpec";
export * from "./schema/ReactInlineContentSpec";
export * from "./schema/ReactStyleSpec";
