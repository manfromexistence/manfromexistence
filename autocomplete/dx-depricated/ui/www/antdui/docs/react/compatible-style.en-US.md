---
group:
  title: Advanced
order: 1
title: CSS Compatible
---

Ant Design supports the last 2 versions of modern browsers. If you need to be compatible with legacy browsers, please perform downgrade processing according to actual needs:

## StyleProvider

Please ref [`@ant-design/cssinjs`](https://github.com/ant-design/cssinjs#styleprovider).

## Compatible adjustment

The CSS-in-JS feature of Ant Design uses the ":where" selector by default to lower the CSS selector specificity, reducing the additional cost of adjusting custom styles when upgrading for users. However, the compatibility of the ":where" syntax is relatively poor in older browsers ([compatibility](https://developer.mozilla.org/en-US/docs/Web/CSS/:where#browser_compatibility)). In certain scenarios, if you need to support older browsers (or encounter priority conflicts like TailwindCSS), you can use `@ant-design/cssinjs` to disable the default lowering of specificity (please ensure version consistency with antd).

```tsx
import { StyleProvider } from '@ant-design/cssinjs';

// Config `hashPriority` to `high` instead of default `low`
// Which will remove `:where` wrapper
export default () => (
  <StyleProvider hashPriority="high">
    <MyApp />
  </StyleProvider>
);
```

It will turn `:where` to class selector:

```diff
--  :where(.css-bAMboO).ant-btn {
++  .css-bAMboO.ant-btn {
      color: #fff;
    }
```

Note: After turning off the `:where` downgrade, you may need to manually adjust the priority of some styles. Or you can **use PostCSS plugin** to raise application css selector priority. PostCSS provides many plugins can help on this. e.g:

- [postcss-scopify](https://www.npmjs.com/package/postcss-scopify)
- [postcss-increase-specificity](https://www.npmjs.com/package/postcss-increase-specificity)
- [postcss-add-root-selector](https://www.npmjs.com/package/postcss-add-root-selector)

Raise priority through plugin:

```diff
--  .my-btn {
++  #root .my-btn {
      background: red;
    }
```

## CSS Logical Properties

To unify LTR and RTL styles, Ant Design uses CSS logical properties. For example, the original `margin-left` is replaced by `margin-inline-start`, so that it is the starting position spacing under both LTR and RTL. If you need to be compatible with older browsers, you can configure `transformers` through the `StyleProvider` of `@ant-design/cssinjs`:

```tsx
import { legacyLogicalPropertiesTransformer, StyleProvider } from '@ant-design/cssinjs';

// `transformers` provides a way to transform CSS properties
export default () => (
  <StyleProvider transformers={[legacyLogicalPropertiesTransformer]}>
    <MyApp />
  </StyleProvider>
);
```

When toggled, styles will downgrade CSS logical properties:

```diff
.ant-modal-root {
-- inset: 0;
++ top: 0;
++ right: 0;
++ bottom: 0;
++ left: 0;
}
```

## Rem Adaptation

In responsive web development, there is a need for a convenient and flexible way to achieve page adaptation and responsive design. The `px2remTransformer` transformer can quickly and accurately convert pixel units in style sheets to rem units relative to the root element (HTML tag), enabling the implementation of adaptive and responsive layouts.

```tsx
import { px2remTransformer, StyleProvider } from '@ant-design/cssinjs';

const px2rem = px2remTransformer({
  rootValue: 32, // 32px = 1rem; @default 16
});

export default () => (
  <StyleProvider transformers={[px2rem]}>
    <MyApp />
  </StyleProvider>
);
```

The resulting transformed styles:

```diff
 .px2rem-box {
-  width: 400px;
+  width: 12.5rem;
   background-color: green;
-  font-size: 32px;
+  font-size: 1rem;
   border: 10PX solid #f0f;
 }

 @media only screen and (max-width: 600px) {
   .px2rem-box {
     background-color: red;
-    margin: 10px;
+    margin: 0.3125rem;
   }
 }
```

### Options

<!-- prettier-ignore -->
| Parameter | Description  | Type | Default |
| --- | --- | --- | --- |
| rootValue | Font size of the root element | `number` | 16 |
| precision | Decimal places for the converted value | `number` | 5 |
| mediaQuery | Whether to convert px in media queries | `boolean` | false |

For more details, please refer to: [px2rem.ts#Options](https://github.com/ant-design/cssinjs/blob/master/src/transformers/px2rem.ts)

## Shadow DOM Usage

Since `<style />` tag insertion is different from normal DOM in Shadow DOM scenario, you need to use `StyleProvider` of `@ant-design/cssinjs` to configure the `container` property to set the insertion position:

```tsx
import { StyleProvider } from '@ant-design/cssinjs';
import { createRoot } from 'react-dom/client';

const shadowRoot = someEle.attachShadow({ mode: 'open' });
const container = document.createElement('div');
shadowRoot.appendChild(container);
const root = createRoot(container);

root.render(
  <StyleProvider container={shadowRoot}>
    <MyApp />
  </StyleProvider>,
);
```
