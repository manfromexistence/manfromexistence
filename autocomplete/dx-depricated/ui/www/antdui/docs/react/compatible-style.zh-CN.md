---
group:
  title: 进阶使用
order: 1
title: 样式兼容
---

Ant Design 支持最近 2 个版本的现代浏览器。如果你需要兼容旧版浏览器，请根据实际需求进行降级处理：

## StyleProvider

查看 [`@ant-design/cssinjs`](https://github.com/ant-design/cssinjs#styleprovider).

## `:where` 选择器

Ant Design 的 CSS-in-JS 默认通过 `:where` 选择器降低 CSS Selector 优先级，以减少用户升级时额外调整自定义样式的成本，不过 `:where` 语法的[兼容性](https://developer.mozilla.org/en-US/docs/Web/CSS/:where#browser_compatibility)在低版本浏览器比较差。在某些场景下你如果需要支持旧版浏览器（或与 [TailwindCSS 优先级冲突](https://github.com/ant-design/ant-design/issues/38794#issuecomment-1328262525)），你可以使用 `@ant-design/cssinjs` 取消默认的降权操作（请注意版本保持与 antd 一致）：

```tsx
import { StyleProvider } from '@ant-design/cssinjs';

// `hashPriority` 默认为 `low`，配置为 `high` 后，
// 会移除 `:where` 选择器封装
export default () => (
  <StyleProvider hashPriority="high">
    <MyApp />
  </StyleProvider>
);
```

切换后，样式将从 `:where` 切换为类选择器：

```diff
--  :where(.css-bAMboO).ant-btn {
++  .css-bAMboO.ant-btn {
      color: #fff;
    }
```

注意：关闭 `:where` 降权后，你可能需要手动调整一些样式的优先级。亦或者**使用 PostCSS 插件**提升应用样式的优先级，PostCSS 提供了非常多的插件来调整优先级，你可以自行按需选择，例如：

- [postcss-scopify](https://www.npmjs.com/package/postcss-scopify)
- [postcss-increase-specificity](https://www.npmjs.com/package/postcss-increase-specificity)
- [postcss-add-root-selector](https://www.npmjs.com/package/postcss-add-root-selector)

通过插件配置，将你的 css 样式进行提升：

```diff
--  .my-btn {
++  #root .my-btn {
      background: red;
    }
```

## CSS 逻辑属性

为了统一 LTR 和 RTL 样式，Ant Design 使用了 CSS 逻辑属性。例如原 `margin-left` 使用 `margin-inline-start` 代替，使其在 LTR 和 RTL 下都为起始位置间距。如果你需要兼容旧版浏览器（如 360 浏览器、QQ 浏览器 等等），可以通过 `@ant-design/cssinjs` 的 `StyleProvider` 配置 `transformers` 将其转换：

```tsx
import { legacyLogicalPropertiesTransformer, StyleProvider } from '@ant-design/cssinjs';

// `transformers` 提供预处理功能将样式进行转换
export default () => (
  <StyleProvider transformers={[legacyLogicalPropertiesTransformer]}>
    <MyApp />
  </StyleProvider>
);
```

切换后，样式将降级 CSS 逻辑属性：

```diff
.ant-modal-root {
-- inset: 0;
++ top: 0;
++ right: 0;
++ bottom: 0;
++ left: 0;
}
```

## rem 适配

在响应式网页开发中，需要一种方便且灵活的方式来实现页面的适配和响应式设计。`px2remTransformer` 转换器可以快速而准确地将样式表中的像素单位转换为相对于根元素（HTML 标签）的 rem 单位，实现页面的自适应和响应式布局。

```tsx
import { px2remTransformer, StyleProvider } from '@ant-design/cssinjs';

const px2rem = px2remTransformer({
  rootValue: 32, // 32px = 1rem; @default 16
});

export default () => (
  <StyleProvider transformers={[px2rem]}>
    <MyApp />
  </StyleProvider>
);
```

最终转换后的样式：

```diff
 .px2rem-box {
-  width: 400px;
+  width: 12.5rem;
   background-color: green;
-  font-size: 32px;
+  font-size: 1rem;
   border: 10PX solid #f0f;
 }

 @media only screen and (max-width: 600px) {
   .px2rem-box {
     background-color: red;
-    margin: 10px;
+    margin: 0.3125rem;
   }
 }
```

### 配置项

<!-- prettier-ignore -->
| 参数 | 说明  | 类型 | 默认值 |
| --- | --- | --- | --- |
| rootValue | 根元素字体大小 | `number` | 16 |
| precision | 转换后的小数点位数 | `number` | 5 |
| mediaQuery | 是否转换媒体查询中的 px | `boolean` | false |

详细请参考: [px2rem.ts#Options](https://github.com/ant-design/cssinjs/blob/master/src/transformers/px2rem.ts)

## Shadow DOM 场景

在 Shadow DOM 场景中，由于其添加 `<style />` 标签的方式与普通 DOM 不同，所以需要使用 `@ant-design/cssinjs` 的 `StyleProvider` 配置 `container` 属性用于设置插入位置：

```tsx
import { StyleProvider } from '@ant-design/cssinjs';
import { createRoot } from 'react-dom/client';

const shadowRoot = someEle.attachShadow({ mode: 'open' });
const container = document.createElement('div');
shadowRoot.appendChild(container);
const root = createRoot(container);

root.render(
  <StyleProvider container={shadowRoot}>
    <MyApp />
  </StyleProvider>,
);
```
