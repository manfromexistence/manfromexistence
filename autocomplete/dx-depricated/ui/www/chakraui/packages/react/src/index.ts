export * from "./components"
export * from "./popper"
export { defaultSystem, defaultConfig } from "./preset"
export { defaultBaseConfig } from "./preset-base"
export * from "./styled-system"
