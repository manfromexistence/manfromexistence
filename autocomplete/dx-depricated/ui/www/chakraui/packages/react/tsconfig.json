{
  "extends": "../../tsconfig.json",
  "include": ["src", "__tests__", "__stories__"]
}
