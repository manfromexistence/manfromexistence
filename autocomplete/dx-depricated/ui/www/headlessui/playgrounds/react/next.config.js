/** @type {import('next').NextConfig} */
module.exports = {
  reactStrictMode: false,
  devIndicators: {
    buildActivity: false,
  },
}
