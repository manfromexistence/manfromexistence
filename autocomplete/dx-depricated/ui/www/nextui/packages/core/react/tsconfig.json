{
  "extends": "../../../tsconfig.json",
  "compilerOptions": {
    "baseUrl": ".",
    "allowJs": true,
    "paths": {
      "tailwind-variants": ["../../../node_modules/tailwind-variants"],
      "@react-aria/interactions": ["../../../node_modules/@react-aria/interactions"]
    }
  },
  "include": ["src", "index.ts"],
  "exclude": ["src/scripts/**", "src/index.ts.backup"]
}
