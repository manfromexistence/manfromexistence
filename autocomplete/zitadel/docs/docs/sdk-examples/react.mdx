---
title: React
sidebar_label: React
---

<table>
  <tr>
    <td width="100px">
      <img width="100px" src="/docs/img/tech/react.png" alt="react" />
    </td>
    <td>
      <a href="https://react.dev/" target="_blank">
        React
      </a>{" "}
      is a declarative, component-based JavaScript library for building user
      interfaces. Integrate authentication to your React application easily by
      using the{" "}
      <a href="https://github.com/zitadel/zitadel-react" target="_blank">
        zitadel-react
      </a>{" "}
      SDK.
    </td>
  </tr>
</table>

### Resources

- [SDK with Example App](https://github.com/zitadel/zitadel-react)
- [Step-By-Step Guide](/docs/examples/login/react)

### React SDK

The [zitadel-react](https://github.com/zitadel/zitadel-react) SDK is a wrapper around the [oidc-client-ts](https://www.npmjs.com/package/oidc-client-ts) and abstracts the handling of specific configurations for ZITADEL.

The following features are covered by the SDK:

- Authenticate your user with ZITADEL using OIDC
- Requesting ZITADEL userinfo endpoint to get user data
- Refresh Token
- Requesting User Roles from userinfo
- Check if user has specified role
- Logout

The goal is to extend the SDK over the time with the following features:

- Integrate ZITADEL APIs to read and manage resources
- Build your own login UI using our Session API

### Example Application

The [zitadel-react](https://github.com/zitadel/zitadel-react) repository also includes an Example Application ready to start and show how a React application looks like with integrated ZITADEL Login.

What does the Example include:

- Home Page with Login Button
- Authenticating user with OIDC PKCE Flow
- Public Page: Accessible without authentication
- Private Page: Shows user information of authenticated user, only accessible after login
- Administrator Page: Only accessible after login and with specific administrator role for the application
- Logout

### Step-By-Step Guide

The [Step-By-Step Guide](/docs/examples/login/react) leads you through the whole process from configuring the right application in ZITADEL to a ready application with integrated Login.

After completing the Step-By-Step Guide you will have:

1. Example Web Application with integrated ZITADEL Login
2. Example page accessible by authenticated user showing retrieved user information
3. Example administrator page accessible by user with administrator role
4. Logout
5. Correct setup for your application in ZITADEL
