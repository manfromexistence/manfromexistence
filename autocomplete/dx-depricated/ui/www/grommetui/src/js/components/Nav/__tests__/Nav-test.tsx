import React from 'react';
import 'jest-styled-components';
import 'jest-axe/extend-expect';
import 'regenerator-runtime/runtime';

import { axe } from 'jest-axe';
import { render } from '@testing-library/react';

import { Grommet, Nav } from '../..';

describe('Nav', () => {
  test('should have no accessibility violations', async () => {
    const { container } = render(
      <Grommet>
        <Nav />
      </Grommet>,
    );

    const results = await axe(container);
    expect(results).toHaveNoViolations();
    expect(container).toMatchSnapshot();
  });
});
