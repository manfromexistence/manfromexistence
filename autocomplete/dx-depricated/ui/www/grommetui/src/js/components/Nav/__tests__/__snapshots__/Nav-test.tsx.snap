// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should have no accessibility violations 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div>
  <div
    class="c0"
  >
    <nav
      class="c1"
    />
  </div>
</div>
`;
